### PR TITLE
Middleware added to redirect user if site expires

### PR DESCRIPTION
**Pull request by marslanabdulrauf**
** Description: **
This PR adds a middle to redirect user to pricing page if the trial site expires

**JIRA:** 
https://edlyio.atlassian.net/browse/EDLY-4882

### CODE REVIEW

File: openedx/features/edly/utils.py [hunk 1]
```diff
@@ -403,6 +403,21 @@ def get_logo_from_current_site_configurations():
...
 
 
+def get_current_plan_from_current_site_configurations():
+    """
+    Gets the "CURRENT_PLAN" value in "DJANGO_SETTINGS_OVERRIDE" from current site configurations.
+
+    Returns:
+        str: current plan.
+    """
+    current_plan = 'essential'  # default value
```
Comment (taimoor-ahmed-1) [inline openedx/features/edly/utils.py:413]: kindly use constant here for `essentials`

File: openedx/features/edly/middleware.py [hunk 2]
```diff
@@ -95,3 +100,16 @@ def _should_update_config(current_value, new_value):
     Check if middleware should replace config value or update it.
     """
     return isinstance(current_value, dict) and isinstance(new_value, dict)
+
+
+def _is_logged_in_path(path):
```
Comment (taimoor-ahmed-1) [inline openedx/features/edly/middleware.py:105]: add docstring

**Review by taimoor-ahmed-1 — CHANGES_REQUESTED** [review]

**Review by taimoor-ahmed-1 — APPROVED** [review]

**Review by taimoor-ahmed-1 — COMMENTED** [review]
A unit test is failing kindly update it.

**Review by taimoor-ahmed-1 — APPROVED** [review]

**Review by madnihamza1841 — APPROVED** [review]
LGTM

**Pull request merged**